src/api/spec/controllers/source_controller_package_command_spec.rb

```diff
@@ -0,0 +1,312 @@
+        package_with_flags.store
+        package_with_flags
+      end
+
+      it 'does not create a new package' do
+        expect { subject }.not_to change(Package, :count)
+      end
+
+      it 'does not change the package flags' do
+        expect { subject }.not_to change(Flag, :count)
+      end
+
+      it 'does not change the package attributes' do
+        expect { subject }.not_to change(package, :title)
+      end
+    end
+
+    context 'without a package' do
+      let(:package) { 'franz' }
+
+      it 'creates a new package' do
+        expect { subject }.to change(Package, :count).from(1).to(2)
+      end
+
+      it 'copies the flags' do
+        expect { subject }.to change(Flag, :count).from(2).to(4)
+      end
+
+      it 'copies the package attributes' do
+        subject
+        expect(project.packages.last.title).to eq('lala')
+      end
+    end
+  end
+
+  # rubocop:disable RSpec/RepeatedExampleGroupBody
+  # rubocop:disable RSpec/RepeatedExample
+  describe 'POST #release' do
+    subject { post :package_command, params: { project: project, package: package, cmd: 'release' } }
+
+    context 'with target_project parameter' do
+      it 'raises MissingParameterError if params[:target_repository].blank? || params[:repository].blank?' do
+        skip('to be tested')
+      end
+
+      it 'does whatever _package_command_release_manual_target does' do
+        skip('to be tested')
+      end
+    end
+
+    context 'without target_project parameter' do
+      it 'does whatever verify_release_targets does' do
+        skip('to be tested')
+      end
+
+      it 'does whatever release_package does' do
+        skip('to be tested')
+      end
+    end
+  end
+
+  describe 'POST #branch' do
+    subject { post :package_command, params: { project: project, package: package, cmd: 'branch' } }
+
+    it 'branches the package' do
+      skip('to be tested')
+    end
+
+    context 'with dryrun' do
+      it 'does nothing' do
+        skip('to be tested')
+      end
+    end
+
+    context 'without permission to create target project' do
+      it 'CreateProjectNoPermission' do
+        skip('to be tested')
+      end
+    end
+
+    context 'without permission to create the target package' do
+      it 'CmdExecutionNoPermission' do
+        skip('to be tested')
+      end
+    end
+  end
+
+  describe 'POST #package_command_fork' do
+    subject { post :package_command, params: { project: project, package: package, cmd: 'fork' } }
+
+    it 'forks the package' do
+      skip('to be tested')
+    end
+
+    context 'without scmsync param' do
+      it 'MissingParameterError' do
+        skip('to be tested')
+      end
+    end
+  end
+
+  describe 'POST #set_flag' do
+    subject { post :package_command, params: { project: project, package: package, cmd: 'set_flag' } }
+
+    it 'adds the flag'
+
+    context 'without flag param' do
+      it 'MissingParameterError' do
+        skip('to be tested')
```

Comment:
```suggestion
      xit 'MissingParameterError'
```
JFYI. You can shortcut the skips like this.